mediorum/server/server_basic_auth.go

```diff
@@ -93,3 +96,60 @@ func (ss *MediorumServer) requireUserSignature(next echo.HandlerFunc) echo.Handl
 		return next(c)
 	}
 }
+
+func (ss *MediorumServer) requireBodySignedByOwner(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		sig, err := signature.ParseFromQueryString(c.QueryParam("signature"))
+		if err != nil {
+			return c.JSON(401, map[string]string{
+				"error":  "invalid signature",
+				"detail": err.Error(),
+			})
+		}
+
+		// hash the body of the POST request
+		bodyBytes, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return c.JSON(401, map[string]string{
+				"error":  "failed to read POST body bytes",
+				"detail": err.Error(),
+			})
+		}
+		c.Request().Body = io.NopCloser(bytes.NewBuffer(bodyBytes)) // reset body so it can be read again later
+		bodyBytesSorted, err := jcs.Transform(bodyBytes)
+		if err != nil {
+			return c.JSON(401, map[string]string{
+				"error":  "failed to sort POST body bytes",
+				"detail": err.Error(),
+			})
+		}
+		bodyHash := crypto.Keccak256Hash(bodyBytesSorted)
+
+		// check signature data is for this request body
+		if !bytes.Equal(bodyHash.Bytes(), sig.DataHash.Bytes()) {
+			return c.JSON(401, map[string]string{
+				"error":  "signature mismatch",
+				"detail": fmt.Sprintf("body hash: %s, signature hash: %s", bodyHash, sig.DataHash),
+			})
+		}
+
+		// check signature not too old
+		age := time.Since(time.Unix(sig.Data.Timestamp/1000, 0))
+		if age > (time.Hour * 48) {
```

Comment:
why 48?

Reply:
just to keep it consistent with the 48 hours we use in the track streaming signature verification. it would also be reasonable to make it much shorter given that this middleware its only used for delisting content currently